Extensions/XMPPMUCSUB/XMPPMUCSUB.h

```diff
@@ -0,0 +1,264 @@
+ *  - It listens for MUCSUB room invitations sent from other users.
+ **/
+@interface XMPPMUCSUB : XMPPModule
+{
+    /*    Inherited from XMPPModule:
+     
+     XMPPStream *xmppStream;
+     
+     dispatch_queue_t moduleQueue;
+     */
+    
+    XMPPIDTracker *xmppIDTracker;
+}
+
+/* Inherited from XMPPModule:
+ 
+ - (id)init;
+ - (id)initWithDispatchQueue:(dispatch_queue_t)queue;
+ 
+ - (BOOL)activate:(XMPPStream *)xmppStream;
+ - (void)deactivate;
+ 
+ @property (readonly) XMPPStream *xmppStream;
+ 
+ - (NSString *)moduleName;
+ 
+ */
+
+/**
+ * Discovering support on MUC service
+ * You can check if MUC/Sub feature is available on MUC service by sending Disco Info IQ
+ * This method will attempt to discover existing services for the domain found in xmppStream.myJID.
+ *
+ * @see xmppMUCSUB:didDiscoverFeatures:
+ * @see xmppMUCSUBFailedToDiscoverFeatures:withError:
+ */
+- (void)discoverFeatures;
+
+/**
+ * This method will attempt to discover support on a specific MUC
+ *
+ * @see xmppMUCSUB:didDiscoverFeaturesForRoom:
+ * @see xmppMUCSUB:failedToDiscoverFeaturesForRoom:withError:
+ *
+ */
+- (BOOL)discoverFeaturesForRoomJID:(XMPPJID *)roomJID;
+
+/**
+ * User can subscribe to  MUC Room events
+ *
+ * Subscription is associated with a nick.It will implicitly register the nick.
+ * Server should otherwise make sure that subscription match the user registered nickname in that room.
+ * In order to change the nick and/or subscription nodes, the same request MUST be sent with a different nick or nodes information.
+ *
+ * A room moderator can subscribe another user to MUC Room events by providing the user JID as an attribute in the <subscribe/> element.
+ *
+ * @see xmppMUCSUB:didSubscribeForRoomJID:
+ * @see xmppMUCSUB:failedToSubscribeForRoomJID:withError:
+ *
+ * @param events the events that user wants to subscribe to.
+ * @param roomJID the room that user wants to subscribe to.
+ * @param userJID Room moderator subscribes another room user.
+ * @param nickName nickname of user in room.
+ * @param password room password if there is any.
+ */
+- (void)subscibeToEvents:(NSArray<XMPPSubscribeEvent> *)events roomJID:(XMPPJID *)roomJID userJID:(XMPPJID *)userJID withNick:(NSString *)nickName passwordForRoom:(NSString *)password;
+
+/**
+ * User can unsubscribe from MUC Room events
+ *
+ * A room moderator can unsubscribe another room user from MUC Room events by providing the user JID as an attribute in the <unsubscribe/> element.
+ *
+ * @see xmppMUCSUB:didUnsubscribeFromRoomJID:
+ * @see xmppMUCSUB:failedToUnsubscribeFromRoomJID:withError:
+ *
+ * @param roomJID the room that user wants to subscribe to.
+ * @param userJID Room moderator unsubscribes another room user.
+ */
+- (void)unsubscibeFromRoomJID:(XMPPJID *)roomJID userJID:(XMPPJID *)userJID;
+
+/**
+ * A user can query the MUC service to get their list of subscriptions.
+ *
+ */
+- (void)fetchSubscriptionList;
```

Comment:
These fetches are async?  Maybe add a comment on how the response gets to the client (e.g. Response will be received by reallyCoolMethod:Parameter: or if there is an error oopsThereWasAnError:Code:

Reply:
yes these are async. Sorry I had forgotten to add call back methods in the comments